contrib/untar.c

```diff
@@ -157,57 +169,58 @@ untar(FILE *a, const char *path)
 		}
 		filesize = parseoct(buff + 124, 12);
 		switch (buff[156]) {
-		case '1':
-			printf(" Ignoring hardlink %s\n", buff);
-			break;
-		case '2':
-			printf(" Ignoring symlink %s\n", buff);
-			break;
-		case '3':
-			printf(" Ignoring character device %s\n", buff);
+			case '1':
+				printf(" Ignoring hardlink %s\n", buff);
+				break;
+			case '2':
+				printf(" Ignoring symlink %s\n", buff);
+				break;
+			case '3':
+				printf(" Ignoring character device %s\n", buff);
+				break;
+			case '4':
+				printf(" Ignoring block device %s\n", buff);
+				break;
+			case '5':
+				printf(" Extracting dir %s\n", buff);
+				create_dir(buff, parseoct(buff + 100, 8));
+				filesize = 0;
+				break;
+			case '6':
+				printf(" Ignoring FIFO %s\n", buff);
+				break;
+			default:
+				printf(" Extracting file %s\n", buff);
+				f = create_file(buff, parseoct(buff + 100, 8));
 				break;
-		case '4':
-			printf(" Ignoring block device %s\n", buff);
-			break;
-		case '5':
-			printf(" Extracting dir %s\n", buff);
-			create_dir(buff, parseoct(buff + 100, 8));
-			filesize = 0;
-			break;
-		case '6':
-			printf(" Ignoring FIFO %s\n", buff);
-			break;
-		default:
-			printf(" Extracting file %s\n", buff);
-			f = create_file(buff, parseoct(buff + 100, 8));
-			break;
 		}
 		while (filesize > 0) {
-			bytes_read = fread(buff, 1, 512, a);
-			if (bytes_read < 512) {
+			bytes_read = fread(buff, 1, BUF_SIZE, a);
+			if (bytes_read < BUF_SIZE) {
 				fprintf(stderr,
-				    "Short read on %s: Expected 512, got %d\n",
-				    path, (int)bytes_read);
+						"Short read on %s: Expected " STR(BUF_SIZE) ", got %"NUM_FORMAT"\n",
+						path, bytes_read);
 				return;
 			}
-			if (filesize < 512)
+			if (filesize < BUF_SIZE)
 				bytes_read = filesize;
 			if (f != NULL) {
 				if (fwrite(buff, 1, bytes_read, f)
-				    != bytes_read)
+					!= bytes_read)
 				{
 					fprintf(stderr, "Failed write\n");
 					fclose(f);
 					f = NULL;
 				}
 			}
-			filesize -= bytes_read;
+			filesize -= (off_t)bytes_read;
 		}
 		if (f != NULL) {
 			fclose(f);
 			f = NULL;
 		}
 	}
+#undef BUF_SIZE
```

Comment:
Style: I dont like this as well, a const int would be much nicer

Reply:
It ensures the definition goes out of scope at the end of the function. A better name might be a good idea, but I can't think of a better way to write this function without relying on new language extensions (which aren't supported by MSVC and other [albeit less popular] compilers).